src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java

```diff
@@ -274,6 +276,43 @@ public void changePassword(@RequestBody PasswordChangeDTO passwordChangeDto) {
         userService.changePassword(passwordChangeDto.getCurrentPassword(), passwordChangeDto.getNewPassword());
     }
 
+    /**
+     * {@code POST  /account/request-api-access} : requests ROLE_API for current user.
+     *
+     * @param apiAccessRequest isRequested boolean and justification (reason for request).
+     */
+    @PostMapping(path = "/account/request-api-access")
+    public void requestApiAccess(@RequestBody ApiAccessRequest apiAccessRequest) {
+        Optional<String> userLogin = SecurityUtils.getCurrentUserLogin();
+        if (userLogin.isPresent()) {
+            Optional<User> user = userService.getUserWithAuthoritiesByLogin(userLogin.get());
+            if (user.isPresent()) {
+                Optional<UserDetailsDTO> userDetailsDTO = userDetailsService.findOneByUser(user.get());
+                if (userDetailsDTO.isPresent()) {
+                    UserDetailsDTO oldUserDetailsDTO = userDetailsDTO.get();
+
+                    AdditionalInfoDTO newAdditionalInfoDTO = oldUserDetailsDTO.getAdditionalInfo();
+                    newAdditionalInfoDTO.setApiAccessRequest(apiAccessRequest);
+
+                    UserDetailsDTO updatedUserDetailsDTO = new UserDetailsDTO();
+                    updatedUserDetailsDTO.setId(oldUserDetailsDTO.getId());
+                    updatedUserDetailsDTO.setLicenseType(oldUserDetailsDTO.getLicenseType());
+                    updatedUserDetailsDTO.setJobTitle(oldUserDetailsDTO.getJobTitle());
+                    updatedUserDetailsDTO.setCompanyName(oldUserDetailsDTO.getCompanyName());
+                    updatedUserDetailsDTO.setCity(oldUserDetailsDTO.getCity());
+                    updatedUserDetailsDTO.setCountry(oldUserDetailsDTO.getCountry());
+                    updatedUserDetailsDTO.setAddress(oldUserDetailsDTO.getAddress());
+                    updatedUserDetailsDTO.setAdditionalInfo(newAdditionalInfoDTO);
+                    updatedUserDetailsDTO.setUserId(oldUserDetailsDTO.getUserId());
+                    updatedUserDetailsDTO.setCompanyId(oldUserDetailsDTO.getCompanyId());
+                    
+                    userDetailsService.save(updatedUserDetailsDTO);
+                    slackService.sendUserApiAccessRequestToChannel(userMapper.userToUserDTO(user.get()));//ask about this, probably slower, but sending to slack so no big deal?
```

Comment:
What does this comment here mean

Reply:
I needed to map the user to a userDTO, which requires a database query, instead of just passing a user. It seems like the slack service mostly relies on the userDTO object. I could change it to just pass the user, but it would not be consistent with the rest of slack service. I figured since it was just sending a slack message, that extra database read might not matter.